src/js/angular/plugins/controllers.js

```diff
@@ -12,22 +12,58 @@ PluginsCtrl.$inject = ['$scope', '$interval', 'toastr', '$repositories', '$licen
 
 function PluginsCtrl($scope, $interval, toastr, $repositories, $licenseService, $modal, $timeout, PluginsRestService, $translate) {
 
+    let timer;
+
+    function cancelTimer() {
+        if (timer) {
+            $interval.cancel(timer);
+        }
+    }
+
     $scope.setPluginIsActive = function (isPluginActive) {
         $scope.pluginIsActive = isPluginActive;
     };
+    const objectsEqual = (o1, o2) =>
+        typeof o1 === 'object' && Object.keys(o1).length > 0
+            ? Object.keys(o1).length === Object.keys(o2).length
+            && Object.keys(o1).every(p => objectsEqual(o1[p], o2[p]))
+            : o1 === o2;
+
     const getPlugins = function () {
-        PluginsRestService.getPlugins($scope.getActiveRepository())
-            .success(function (data) {
-                $scope.plugins = $scope.buildPluginsArray(data.results.bindings);
-                if (angular.isDefined($scope.plugins)) {
-                    $scope.displayedPlugins = $scope.plugins;
-                }
-                $scope.matchedElements = $scope.plugins;
-            }).error(function (data) {
-            toastr.error(getError(data));
-        }).finally(function () {
-            $scope.setLoader(false);
-        });
+        timer = $interval(function () {
+
+            PluginsRestService.getPlugins($scope.getActiveRepository())
```

Comment:
This code block is almost identical with the code in initPlugins? Do you really need them both? Is it possible to somehow reuse the logic?